### PR TITLE
Add ghc-bin

### PR DESCRIPTION
**Pull request by depsterr**
## Description of package

GHC - The Glourious Glasgow Haskell Compilation System

This is the official binary release for x86_64 Alpine Lunix which uses musl libc

## Installed manifest of package

Too large for pr, [pastebin](https://pastebin.com/APEiq3BL)

## New package

- [x] Latest upstream version.
- [x] I agree to maintain this package.
- [x] I agree to notify @dylanaraps if I can no longer maintain this package.

### CODE REVIEW

**Comment by depsterr** [comment]
It shows the commits from my previous PR in the commit history for some reason, but the only new addition here is ghc-bin

**Comment by depsterr** [comment]
I just noticed files not ready for installing, I will attempt to fix the package and then reopen. Apologies.

**Comment by depsterr** [comment]
The Makefile doesn't support DESTDIR and I don't have the energy to rewrite it / reverse engineer it now so this will probably remain closed.

**Comment by dylanaraps** [comment]
I'm looking into it, this will be handy to have in Community.

File: community/ghc-bin/depends [hunk 1]
```diff
@@ -0,0 +1,2 @@
+libffi
```
Comment (dylanaraps) [inline community/ghc-bin/depends:1]: Missing a dependency on `gmp` as well.

Comment (depsterr) [reply community/ghc-bin/depends:1]: From the official download page:

> Linux (x86-64)
> Alpine
> Alpine Linux 3.10 for x86-64. This is a complete build, including interactive system, profiling libraries and documentation. Unlike our other binary distributions, this links against the integer-simple big-integer backend and therefore does not require libgmp.

**Comment by depsterr** [comment]
It is worth mentioning that the first approach of simply copying the files in bin, lib, and include worked, just that it copied a large amount of unecessary files as well.

There is also of course the provided makefile, but as I said, setting DESTDIR doesn't seem to work (even though it is set in one of the makefile config files).

The easiest way would probably be creating a custom Makefile or copying the files manually.

I'll do a `find /` before and after installing without kiss and diff the results to see if I can figure out what files are needed.

**Comment by depsterr** [comment]
After investigating, it seems that the Makefile actually does exactly what my install script did, I just thought is seemed off. The number of installed files is exactly the same and after examining parts of the content it seems to match.

Also the reason that ghci is provided as a file is that otherwise it isn't recognised by kiss manifest and therefore wouldn't be removed upon package removal. Otherwise it's just the script provided by ghc but fixed to pass shellcheck.

Since everything seems to be working I will reopen the pr.

**Comment by dylanaraps** [comment]
Nice. Just one thing:

```sh
# These are provided by libffi.
rm -f "$1/usr/lib/libffi.so" \
      "$1/usr/include/ffitarget.h" \
      "$1/usr/include/ffi.h"
```

This will prevent GHC from providing a duplicate `libffi`.

**Comment by depsterr** [comment]
Done!

**Comment by dylanaraps** [comment]
It also seems to link to GMP from `/lib`?

```sh
        libgmp.so.10 => /lib/libgmp.so.10 (0x7f47f4c94000)
```

Also, it links to `ncurses` so it should probably be added as a dependency.

**Comment by depsterr** [comment]
ncurses is a dependency already, but I'll add gmp since it links to it. Weird that it would link to it even though their page said it wasn't a dependency though.

**Comment by dylanaraps** [comment]
Yeah. It is odd. I get a bunch of linker errors without `gmp` for reference:

```
Error relocating /usr/bin/../lib/x86_64-linux-ghc-8.10.1/libHSinteger-gmp-1.0.3.0-ghc8.10.1.so: __gmpn_tdiv_qr: symbol not found
Error relocating /usr/bin/../lib/x86_64-linux-ghc-8.10.1/libHSinteger-gmp-1.0.3.0-ghc8.10.1.so: __gmpn_gcd_1: symbol not found
Error relocating /usr/bin/../lib/x86_64-linux-ghc-8.10.1/libHSinteger-gmp-1.0.3.0-ghc8.10.1.so: __gmpn_add_1: symbol not found
Error relocating /usr/bin/../lib/x86_64-linux-ghc-8.10.1/libHSinteger-gmp-1.0.3.0-ghc8.10.1.so: __gmpn_andn_n: symbol not found
Error relocating /usr/bin/../lib/x86_64-linux-ghc-8.10.1/libHSinteger-gmp-1.0.3.0-ghc8.10.1.so: __gmpz_gcdext: symbol not found
Error relocating /usr/bin/../lib/x86_64-linux-ghc-8.10.1/libHSinteger-gmp-1.0.3.0-ghc8.10.1.so: __gmpn_lshift: symbol not found
Error relocating /usr/bin/../lib/x86_64-linux-ghc-8.10.1/libHSinteger-gmp-1.0.3.0-ghc8.10.1.so: __gmpn_rshift: symbol not found
Error relocating /usr/bin/../lib/x86_64-linux-ghc-8.10.1/libHSinteger-gmp-1.0.3.0-ghc8.10.1.so: __gmpn_and_n: symbol not found
Error relocating /usr/bin/../lib/x86_64-linux-ghc-8.10.1/libHSinteger-gmp-1.0.3.0-ghc8.10.1.so: __gmpz_powm: symbol not found
Error relocating /usr/bin/../lib/x86_64-linux-ghc-8.10.1/libHSinteger-gmp-1.0.3.0-ghc8.10.1.so: __gmpn_mod_1: symbol not found
Error relocating /usr/bin/../lib/x86_64-linux-ghc-8.10.1/libHSinteger-gmp-1.0.3.0-ghc8.10.1.so: __gmpn_cmp: symbol not found
Error relocating /usr/bin/../lib/x86_64-linux-ghc-8.10.1/libHSinteger-gmp-1.0.3.0-ghc8.10.1.so: __gmpn_add: symbol not found
Error relocating /usr/bin/../lib/x86_64-linux-ghc-8.10.1/libHSinteger-gmp-1.0.3.0-ghc8.10.1.so: __gmpn_sub_1: symbol not found
Error relocating /usr/bin/../lib/x86_64-linux-ghc-8.10.1/libHSinteger-gmp-1.0.3.0-ghc8.10.1.so: __gmpn_xor_n: symbol not found
Error relocating /usr/bin/../lib/x86_64-linux-ghc-8.10.1/libHSinteger-gmp-1.0.3.0-ghc8.10.1.so: __gmpz_clear: symbol not found
Error relocating /usr/bin/../lib/x86_64-linux-ghc-8.10.1/libHSinteger-gmp-1.0.3.0-ghc8.10.1.so: __gmpz_invert: symbol not found
```

**Comment by dylanaraps** [comment]
Thanks!

**Comment by depsterr** [comment]
No problem! I'll message the upstream about gmp and see what they have to say about it.

**Pull request merged**